packages/remote-config/src/abt/experiment.ts

```diff
@@ -45,32 +50,37 @@ export class Experiment {
     currentActiveExperiments: Set<string>,
     experimentInfoMap: Map<string, FirebaseExperimentDescription>
   ): void {
+    const customProperty: Record<string, string | null> = {};
     for (const [experimentId, experimentInfo] of experimentInfoMap.entries()) {
       if (!currentActiveExperiments.has(experimentId)) {
-        this.addExperimentToAnalytics(experimentId, experimentInfo.variantId);
+        customProperty[experimentId] = experimentInfo.variantId;
       }
     }
+    void this.addExperimentToAnalytics(customProperty);
   }
 
   private removeInactiveExperiments(
     currentActiveExperiments: Set<string>,
     experimentInfoMap: Map<string, FirebaseExperimentDescription>
   ): void {
+    const customProperty: Record<string, string | null> = {};
     for (const experimentId of currentActiveExperiments) {
       if (!experimentInfoMap.has(experimentId)) {
-        this.removeExperimentFromAnalytics(experimentId);
+        customProperty[experimentId] = null;
       }
     }
+    void this.addExperimentToAnalytics(customProperty);
   }
 
-  private addExperimentToAnalytics(
-    _experimentId: string,
-    _variantId: string
-  ): void {
-    // TODO
-  }
-
-  private removeExperimentFromAnalytics(_experimentId: string): void {
-    // TODO
+  private async addExperimentToAnalytics(
+    customProperty: Record<string, string | null>
+  ): Promise<void> {
+    try {
+      const analytics = await this.analyticsProvider.get();
```

Comment:
So I think we talked about this before, but being able to use this requires the developer to have called `import { anything } from 'firebase/analytics'`. If they don't, this line will hang. I'm actually a little concerned about the messaging implementation (the pattern you took this from) running into this. I think the best way to handle this might be to use `getImmediate({ optional: true })` instead. Example: https://github.com/firebase/firebase-js-sdk/blob/main/packages/storage/src/service.ts#L286

This returns undefined if the developer did not import 'firebase/analytics'. You could use that to log a warning message, or continue if it was found. You'd also want to make it clear in documentation that users that use this feature must also import and initialize analytics.

Reply:
I have updated to include a warning message